src/ucp/api/ucp.h

```diff
@@ -439,6 +440,19 @@ enum {
 };
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief UCP memory mapping protection mode.
+ *
+ * The enumeration list describes the memory mapping protections supported by @ref
+ * ucp_mem_map() function.
+ */
+enum {
+    UCP_MEM_MAP_PROT_READ   = UCS_BIT(0), /**< Enable read access. */
```

Comment:
I would like to withdraw my approval based on UCT patch review :) Let's add LOCAL to both - enum name and description. @yosefe Maybe you want to use this opportunity to add remote read/write ?

Reply:
UCT patch #5348

Reply:
@shamisp do you mean the flags we need are: UCP_MAM_MAP_PROT_[LOCAL|REMOTE]_[READ|WRITE] ?

Reply:
I think so. Even if you want to define only `local` for now, I would explicitly name it `_local_`.

Reply:
Also, to some extend this might be a hint, since some transports may not have permissions for read/write, etc.

Reply:
@shamisp Thanks for the comment! As our plan  for example UCP_MEM_MAP_PROT_READ would translate to UCT_MD_MEM_ACCESS_REMOTE_GET| UCT_MD_MEM_ACCESS_LOCAL_READ. So we don't need adding LOCAL to UCP enums.

Reply:
@shamisp @yosefe Per original plan, we were going to translate UCP_MEM_MAP_PROT_READ to UCT_MD_MEM_ACCESS_REMOTE_GET| UCT_MD_MEM_ACCESS_LOCAL_READ.

Reply:
> @shamisp @yosefe Per original plan, we were going to translate UCP_MEM_MAP_PROT_READ to UCT_MD_MEM_ACCESS_REMOTE_GET| UCT_MD_MEM_ACCESS_LOCAL_READ.

We can add "convenience" in UCP:
 UCP_MEM_MAP_PROT_READ =  UCP_MEM_MAP_PROT_LOCAL_READ| UCP_MEM_MAP_PROT_REMOTE_READ

Reply:
@shamisp @yosefe Fixed. One more look pls. Just one more thing to confirm, we are using REMOTE_GET and REMOTE_PUT in UCT, we can use REMOTE_READ and REMOTE_WRITE in UCP. Right?

Reply:
@yosefe I don't understand what is the value of pre-defined UCP_MEM_MAP_PROT_READ = UCP_MEM_MAP_PROT_LOCAL_READ| UCP_MEM_MAP_PROT_REMOTE_READ

Reply:
syntax sugar

Reply:
IMHO, it looks completely out of place in UCP.H. Extra enum to document and manage with no additional value.

Reply:
@shamisp @yosefe  I removed the extra enum.